ipn/ipnlocal/c2n.go

```diff
@@ -149,6 +150,10 @@ func (b *LocalBackend) handleC2NUpdate(w http.ResponseWriter, r *http.Request) {
 		res.Err = "not supported"
 		return
 	}
+	if prefs.Check && !prefs.Apply {
+		// Client opted in to update notifications but not automatic updates.
+		return
```

Comment:
note line 145-147 above. It's already bailing out based on the prefs you set on line 133.